lambda/chaos.py

```diff
@@ -0,0 +1,306 @@
+import json
+import boto3
+import os
+import time
+import requests
+import time
+import json
+import random
+
+ssm = boto3.client('ssm')
+fis = boto3.client('fis')
+
+FAULTS=['emqx-node-shutdown',
+        'emqx-high-cpu-100',
+        'emqx-high-cpu-80',
+        'emqx-high-io-80',
+        'emqx-high-io-100',
+        'emqx-kill-proc',
+        'emqx-high-mem-80',
+        'emqx-high-mem-95',
+        'emqx-distport-blackhole',
+        'emqx-latency-200ms',
+        'emqx-packet-loss-10',
+        'kafka-plaintext-latency-200',
+        'kafka-plaintext-pktloss-10',
+        'kafka-plaintext-pktloss-100']
+
+# lambda: Catch and Retry
+class Retry(Exception):
+    pass
+# lambda: Catch but unrecoverable
+#
+class Fail(Exception):
+    pass
+
+'''
+Run system manager command
+'''
+
+def run_cmd_on(cluster_name : str, command_name : str, services : list,
+               cmd_parms : dict() = {}):
+    doc_name='-'.join([cluster_name, command_name])
+    print(f"run {command_name} ...")
+    res = ssm.send_command(DocumentName=doc_name, TimeoutSeconds=30,
+                           Targets=[{"Key":"tag:cluster","Values":[cluster_name]},
+                                    {"Key":"tag:service","Values":services}],
+                           Parameters=cmd_parms
+                           )
+    """ example of return
+    {'Command': {'CommandId': 'bf834b53-0507-4e94-812d-725366985d69',
+                 'DocumentName': 'stop_traffic-william-k2',
+                 'DocumentVersion': '$DEFAULT',
+                 'Comment': '',
+                 'ExpiresAfter': datetime.datetime(2022, 2, 16, 15, 19, 22, 524000, tzinfo=tzlocal()),
+                 'Parameters': {},
+                 'InstanceIds': [],
+                 'Targets': [{'Key': 'tag:cluster', 'Values': ['william-k2']}, {'Key': 'tag:service', 'Values': ['loadgen']}],
+                 'RequestedDateTime': datetime.datetime(2022, 2, 16, 14, 18, 52, 524000, tzinfo=tzlocal()),
+                 'Status': 'Pending', 'StatusDetails': 'Pending',
+                 'OutputS3Region': 'eu-west-1', 'OutputS3BucketName': '',
+                 'OutputS3KeyPrefix': '', 'MaxConcurrency': '50', 'MaxErrors': '0',
+                 'TargetCount': 0, 'CompletedCount': 0, 'ErrorCount': 0, 'DeliveryTimedOutCount': 0,
+                 'ServiceRole': '', 'NotificationConfig': {'NotificationArn': '', 'NotificationEvents': [], 'NotificationType': ''},
+                 ....
+    """
+    return res
+
+def find_exp_id(cluster_name : str, fault_name : str, next_token=None) -> str:
+    if next_token:
+        res=fis.list_experiment_templates(maxResults=100, nextToken=next_token)
+    else:
+        res=fis.list_experiment_templates(maxResults=100)
+    for t in res['experimentTemplates']:
+        if t['tags']['cluster'] == cluster_name and t['tags']['fault_name'] == fault_name:
+            return t['id']
+    if 'nextToken' in res:
+        find_exp_id(cluster_name, fault_name, next_token=res['nextToken'])
```

Comment:
```suggestion
        return find_exp_id(cluster_name, fault_name, next_token=res['nextToken'])
```

Also, I think that Python does not handle tail recursion very well (although we'll certainly not have enough pages to trigger that :joy_cat: )